google/cloud/forseti/common/gcp_api/storage.py

```diff
@@ -686,3 +692,31 @@ def get_object_iam_policy(self, bucket, object_name, user_project=None):
                 'objectIamPolicy', e, 'bucket', bucket)
             LOGGER.error(api_exception)
             raise api_exception
+
+
+def upload_json(data, gcs_upload_path):
+    """Upload inventory summary/violation data in json format.
+
+    Args:
+        data (dict): the data to upload
+        gcs_upload_path (string): the GCS upload path.
+    """
+    with tempfile.NamedTemporaryFile() as tmp_violations:
+        tmp_violations.write(parser.json_stringify(data))
+        tmp_violations.flush()
+        storage_client = StorageClient()
+        storage_client.put_text_file(tmp_violations.name, gcs_upload_path)
```

Comment:
What filename is being given here?

Reply:
I don't understand the question. Sorry.

Reply:
I was looking at `tmp_violations.name`, and then wondering what is the resulting file name.